astropy/stats/lombscargle/core.py

```diff
@@ -356,3 +356,97 @@ def model(self, t, frequency):
                              fit_mean=self.fit_mean,
                              nterms=self.nterms)
         return y_fit * get_unit(self.y)
+
+    def cumulative_distribution(self, z, normalization='standard'):
+        """
+        Compute the theoretical cumulative distribution of the periodogram
+        for the null hypothesis – i.e. data consisting of Gaussian noise.
+
+        Parameters
+        ----------
+        z : array-like
+            the periodogram value
+        normalziation : string
+            The periodogram normalization. Must be one of
+            ['standard', 'model', 'log', 'psd']
+
+        Returns
+        -------
+        cdf : np.ndarray
+            The theoretical cumulative distribution function
+
+        Notes
+        -----
+        Expressions used here are adapted from Baluev 2008, Table 1.
+        """
+        # TODO: handle units in z?
+        z = np.asarray(z)
```

Comment:
I would really just advocate removing this coercion -- the code below will already fail with a `UnitsError` is `z` is not dimensionless, and will give the correct answer if, e.g., a user passes in `z=np.arange(0,100) * u.percent`. In contrast, for that case with the `np.asarray` clause the returned values are wrong and with @pllim's suggestion, one gets an error for no reason.

More generally, I think that unless you *know* that the code will give the wrong answer for a given type of input, you should *not* do any tests on the input (but then, I really like the ducktyping aspect of python and I've learned over the last years that `Quantity` is amply mature enough that, generally, you simply do not have to worry about using it).